app/server_platform.py

```diff
@@ -36,33 +32,31 @@ def _params_to_pairs(params: list[Pair]) -> list[tuple[str, str]]:
     return [(param.name, param.value) for param in params] if params else []
 
 
+def _read_key_password(key_path: str) -> str | None:
+    return os.environ.get(re.sub("[\-\.\/]", "_", key_path).upper() + "_PASSWORD")
+
+
 class NoRedirectHandler(HTTPRedirectHandler):
     def redirect_request(self, req, fp, code, msg, headers, newurl):
         logging.debug("Got redirect")
         return None
 
 
-# For the future: extend this class from enablebanking's platform
 class ServerPlatform:
-    PATH_PREFIX = "/app/open_banking_certs/"
+    OB_CERTS_DIR = os.environ.get("OB_CERTS_DIR", "/app/open_banking_certs")
 
-    def update_tls_paths(self, tls: TLS):
-        fields = ["cert_path", "key_path", "ca_cert_path"]
-        for field in fields:
-            if hasattr(tls, field):
-                field_value = getattr(tls, field)
-                if field_value and not field_value.startswith(self.PATH_PREFIX):
-                    setattr(tls, field, self.PATH_PREFIX + field_value)
+    hazmat_backend = default_backend()
 
     def get_ssl_context(self, tls: TLS | None) -> ssl.SSLContext:
         if tls:
-            self.update_tls_paths(tls)
             ssl_context = ssl.create_default_context()
             ssl_context.load_cert_chain(
-                tls.cert_path, tls.key_path, lambda: tls.key_password
+                os.path.join(self.OB_CERTS_DIR, tls.cert_path),
+                os.path.join(self.OB_CERTS_DIR, tls.key_path),
```

Comment:
I think that passing something like "../../app/broker_tls/server.pem" will allow us making calls on behalf of arbitrary certificates/keys

Reply:
yes, I was thinking the same. I believe the problem (if it's a problem) was always there, but I believe it's easy to fix by checking that the paths are relative to OB_CERTS_DIR. I'll add the check

Reply:
I think os.path.abspath can be used together with checking that the result starts with `OB_CERTS_DIR`